src/mappings/handlers/evmHandlers.ts

```diff
@@ -125,3 +428,56 @@ async function _handleEvmTransfer(event: TransferLog): Promise<void> {
     await txOut.save()
   }
 }
+
+async function updateLoans(poolId: string, blockDate: Date, shelf: string, pile: string, navFeed: string) {
+  let loanIndex = 1
+  const contractLoans = []
+  // eslint-disable-next-line
+  while (true) {
+    const shelfContract = Shelf__factory.connect(shelf, ethApi)
+    const response = await shelfContract.token(loanIndex)
```

Comment:
For this, couldn't we get the # of loans that already exist, and initiate `loanIndex` not at 1 but rather at the known # of loans? Would save a lot of calls I think

Reply:
Basically just initiate `loanIndex = existingLoans.length` from below I think